File: 003.Haskell/uri.hs
main :: IO ()
main = do
        split :: Eq a => a -> [a] -> [[a]] -- Web
        split d [] = []
        split d s = x : split d (drop 1 y) where (x,y) = span (/= d) s
        
        t <- readLn :: IO [Char]
        [e, d] = split " " t
        
        putStrLn $ id ("X = " ++ show(a + b))


        -- e, d = input().split(" ")
        -- e, d = int(e), int(d)
        
        -- if(e <= d-3):
        --     print("Muito bem! Apresenta antes do Natal!")
        -- elif(e > d):
        --     print("Eu odeio a professora!")
        -- else:
        --     print("Parece o trabalho do meu filho!")
        --     e += 2
        --     if(e < 24):
        --         print("TCC Apresentado!")
        --     else:
        --         print("Fail! Entao eh nataaaaal!")

File: 002.Python/uri.py
input()
lista = list(map(int, input().split()))
t = 0

if len(lista) == 1:
  t = 1
else:
  if lista[0] > lista[1]:
    for i in range(1, (len(lista) - 1)):
      if i % 2 != 0:
        if not(lista[i] < lista[i+1]):
          t = 0
          break
        else:
          t = 1
  elif lista[0] < lista[1]:
    for i in range(1, (len(lista) - 1)):
      if i % 2 != 0:
        if not(lista[i] > lista[i+1]):
          t = 0
          break
        else:
          t = 1
  else:
    t = 0

print(t)